### PR TITLE
Flow Metrics: Docs

### PR DESCRIPTION
**Pull request by yaauie**
## Release notes

[rn:skip]

## What does this PR do?

Adds documentation for the per-pipeline `flow` metrics that are being exposed in the Node Stats API

## Why is it important/What is the impact to the user?

This helps our users understand what the numbers in the `flow` metrics mean.

## Checklist

- ~[ ] My code follows the style guidelines of this project~
- ~[ ] I have commented my code, particularly in hard-to-understand areas~
- [x] I have made corresponding changes to the documentation
- ~[ ] I have made corresponding change to the default configuration files (and/or docker env variables)~
- ~[ ] I have added tests that prove my fix is effective or that my feature works~

### CODE REVIEW

File: docs/static/monitoring/monitoring-apis.asciidoc [hunk 1]
```diff
@@ -462,6 +462,10 @@ The following request returns a JSON document containing pipeline stats,
 including:
 
 * the number of events that were input, filtered, or output by each pipeline
+* the current and lifetime _flow_ rates for useful pipeline metrics:
+** `input_throughput`, `filter_throughput`, and `output_throughput`, expressed in _events-per-second_
+** `worker_concurrency`, expressed as a ratio of worker execution-time to wall-clock time
+** `queue_backpressure`, expressed as a ratio of time inputs are blocked pushing to the queue to wall-clock time (_NOTE: magnitude of this flow rate is heavily dependent on pipeline structure and is not comparable across pipelines_)
```
Comment (yaauie) [inline docs/static/monitoring/monitoring-apis.asciidoc:468]: @karenzone I am not sure here is the best place to document the _meaning_ of our flow-metrics long-term; it is certainly more detail than any of the other bullet-points in the list.

Can you help me find a better home for this granularity?

For context, this is in the scope of #14463, in which we introduce "flow" metrics to represent rates-of-change in addition to the current cumulative-value, to make it easier to see how a pipeline is performing _now_ relative to how it has been performing _before_.
 - In the initial release, we are first adding 5 "flow" metrics to _pipelines_, each of which are the rate of change of a particular pipeline metric with respect to time passing, and each of the flow metrics will present a `lifetime` rate, and a `current` rate.
 - Subsequent iterations will add more rolling averages to each flow metric (so we'll continue to have `current` and `lifetime`, but will also have something like `last_1h` once the pipeline has been alive for more than an hour; the precise periods that are covered by these rolling averages is to-be-determined and will likely come with a Technology Preview while we settle on the "right" granularity)
 - Subsequent iterations will also add flow metrics to other things, like PQ metrics, the process as a whole, and even individual plugins.
 - Future flow metrics that we add _may_ use something other than (wall-clock) time as their denominator, such as an individual plugin's throughput using the amount of time that particular plugin has spent in execution.

File: docs/static/monitoring/monitoring-apis.asciidoc [hunk 2]
```diff
@@ -487,6 +491,28 @@ Example response:
...
+        "filter_throughput" : {
+          "current": 187.810,
+          "lifetime": 201.799
+        },
+        "output_throughput" : {
+          "current": 191.087,
+          "lifetime": 201.761
+        },
+        "queue_backpressure" : {
+          "current": 0.277,
```
Comment (mashhurs) [inline docs/static/monitoring/monitoring-apis.asciidoc:508]: can we also mention the range particularly for backpressure? I feel users can set up a threshold to get the alarms. Or if we will have a chance to integrate in APM, worth mentioning alarm feature (ES watchers may be helpful?!)

Comment (yaauie) [reply docs/static/monitoring/monitoring-apis.asciidoc:508]: We most certainly need to document it.

For pipelines that are being _pushed_ data, effectively the only good number is zero.

The magnitude of value is highly dependent on the shape of the pipeline's inputs, and is _not_ comparable from one pipeline to another. A pipeline with a single-threaded input (like stdin, pipeline) will have a range of 0-1 (what percent of wall-clock time is that one thread blocked), while a pipeline with inputs that have many inbound connections will have a much higher range (what percentage of wall-clock time are _each_ of those connections blocked, _added together_ -- 100 inbound connections that are each blocked 75% of the time will have a value of about `75.0`).

Additionally, with inputs that are _pulling_ data (local file input, jdbc input, elasticsearch input, http poller, etc), some amount of queue back-pressure is normal and expected, and results in the input pulling data a little slower but _without losing its place_.

**Comment by yaauie** [comment]
This effort was superseded by #14539

**Pull request closed without merging**